src/components/ViewDate.jsx

```diff
@@ -19,17 +20,28 @@ class ViewDate extends React.Component {
     this.state = { open: false }
   }
 
-  weekNum () {
-    return moment(this.props.viewDate).format('Wo')
-  }
+  weekParity (weekParity) {
+    if (weekParity !== 'even' && weekParity !== 'odd') {
+      return '-'
+    }
 
-  weekParity () {
-    const parity = moment(this.props.viewDate).isoWeek() % 2 === 0 ? 'even' : 'odd'
-    return CP.translate(parity)
+    return CP.translate(weekParity)
   }
 
   render () {
-    return <div className="view-date">{CP.translate('weekNav.teachweek', { weeknum: this.weekNum() })} ({this.weekParity()})</div>
+    const weekParity = this.weekParity(this.props.weekParity)
+    const translationExists = R.contains(this.props.weekType, ['teaching', 'exams', 'holiday'])
```

Comment:
This is quite not performant. Can't we just pass in the fallback to the CP.transpate function?